src/waitress/buffers.py

```diff
@@ -224,13 +222,11 @@ def __len__(self):
         else:
             return self.strbuf.__len__()
 
-    def __nonzero__(self):
+    def __bool__(self):
         # use self.__len__ rather than len(self) FBO of not getting
```

Comment:
Presumably we don't need `__bool__` either:

```
>>> class Foo:
...     def __init__(self, x): self.x = x
...     def __len__(self): return self.x
... 
>>> bool(Foo(1))
True
>>> bool(Foo(5))
True
>>> bool(Foo(0))
False
```

Reply:
That might have consequences which I am not able to judge.

Skimming the code it seems that now you can have a length `FileBasedBuffer` instance which is `True`.
Removing the `__bool__` method this will change to `False`.

But, as said, I am not much in to this code to say if thic can be a good or bad thing.